### ISSUE TITLE
current project : Cannot read property 'timestamp' of undefined

### ISSUE THREAD

**Issue by JulienLamy76** [isomorphic-git/isomorphic-git#721]
Hi, I'm using node for isomorphic.

When I'm using the git log like this :
```js
isomorphic.log(
{
    fs: this.electronService.fs,
    dir: this.path
}).then((result) => {
    console.log(result);
}).catch((err) => {
    console.log(err);
});
```

It works perfectly for my projects, but when I'm doing this on my current project, I've this error :
TypeError: Cannot read property 'timestamp' of undefined
    at compareAge (index.js:6402)
    at Array.sort (:4200/native)
    at Module.log$1 (index.js:6493)

And I don't know why i'm getting this.

**Comment by jcubic** [comment]
Try using fs as plugin, not sure if old way was removed, check documentation.

**Comment by JulienLamy76** [comment]
By doing this :
```js
isomorphic.plugins.set('fs', this.electronService.fs);
isomorphic.log(
    {
    dir: this.path
    }).then((result) => {
    console.log(result);
    }).catch((err) => {
    console.log(err);
    });
```
I've the same error and one more time, only on my current project not on others.
(By the way, yes, I'm using electron, it's why I've the fs in my electronService)

**Comment by jcubic** [comment]
and `this.electronService.fs` works when you pass it like this? Did you tried using it without isomorphic git? If it work I don't know what is the reason, maybe @wmhilton will need to debug in Electron. I don't know if it's supported platform.

**Comment by JulienLamy76** [comment]
Yes it works with another example when I'm getting the last commit and works also for git log I'm doing on every repo except this one (the current one).

**Comment by jcubic** [comment]
If it works for other projects in Electron, then you need to figure out what is the difference. Try opening google chrome dev tools and enable pause on exception i source tab, and investigate the variables.

**Comment by wmhilton** [comment]
> Try using fs as plugin, not sure if old way was removed, check documentation.

Nope, that options hasn't been removed yet! But the plugin method is definitely preferred.

```
TypeError: Cannot read property 'timestamp' of undefined
at compareAge (index.js:6402)
at Array.sort (:4200/native)
at Module.log$1 (index.js:6493)
```

it could be that `logCommit` is returning an object in the format `{oid, error}`which would mean `committer` is undefined (hence "property 'timestamp' of undefined).

I'm checking for that [here](https://github.com/isomorphic-git/isomorphic-git/blob/dc16a6cb34bf58d19886e3183851bf3c1da9e125/src/commands/log.js#L39) but it looks like I forgot to check [here](https://github.com/isomorphic-git/isomorphic-git/blob/dc16a6cb34bf58d19886e3183851bf3c1da9e125/src/commands/log.js#L62)... I can't imagine a scenario where you wouldn't have the parent commits of a non-shallow commit though......

>  As if like a commit or author is creating the bug...

Oooh, that could be it! Maybe something about that particular commit fools the commit parser, and so it _is_ returning a commit (not an `{oid, error}` object) but the parsed commit is missing the committer (and committer timestamp)? What commit is it? Was the commit made using isomorphic-git? (I wonder if I allow creating invalid commits if you pass in an empty object for `committer` or something.)
